src/bin/solang.rs

```diff
@@ -507,106 +566,106 @@ fn process_file(
-            continue;
-        }
+    let resolved_contract = &ns.contracts[contract_no];
 
-        if let Some("cfg") = matches.get_one::<String>("EMIT").map(|v| v.as_str()) {
-            println!("{}", resolved_contract.print_cfg(&ns));
-            continue;
-        }
+    if !resolved_contract.instantiable {
+        return;
+    }
 
-        if verbose {
-            if target == solang::Target::Solana {
-                eprintln!(
-                    "info: contract {} uses at least {} bytes account data",
-                    resolved_contract.name, resolved_contract.fixed_layout_size,
-                );
-            }
+    if let Some("cfg") = matches.get_one::<String>("EMIT").map(|v| v.as_str()) {
+        println!("{}", resolved_contract.print_cfg(ns));
+        return;
+    }
 
+    if verbose {
+        if ns.target == solang::Target::Solana {
             eprintln!(
-                "info: Generating LLVM IR for contract {} with target {}",
-                resolved_contract.name, ns.target
+                "info: contract {} uses at least {} bytes account data",
+                resolved_contract.name, resolved_contract.fixed_layout_size,
             );
         }
 
-        let context = inkwell::context::Context::create();
-        let filename_string = filename.to_string_lossy();
+        eprintln!(
+            "info: Generating LLVM IR for contract {} with target {}",
+            resolved_contract.name, ns.target
+        );
+    }
 
-        let binary = resolved_contract.binary(&ns, &context, &filename_string, opt);
+    let context = inkwell::context::Context::create();
+    let filename_string = filename.to_string_lossy();
 
-        if save_intermediates(&binary, matches) {
-            continue;
-        }
+    let binary = resolved_contract.binary(ns, &context, &filename_string, opt);
 
-        let code = binary.code(Generate::Linked).expect("llvm build");
+    if save_intermediates(&binary, matches) {
+        return;
+    }
 
-        if std_json {
-            json_contracts.insert(
-                binary.name.to_owned(),
-                JsonContract {
-                    abi: abi::ethereum::gen_abi(contract_no, &ns),
-                    ewasm: Some(EwasmContract {
-                        wasm: hex::encode_upper(code),
-                    }),
-                    minimum_space: None,
-                },
-            );
-        } else {
-            let bin_filename = output_file(matches, &binary.name, target.file_extension());
+    let code = binary.code(Generate::Linked).expect("llvm build");
 
-            if verbose {
-                eprintln!(
-                    "info: Saving binary {} for contract {}",
-                    bin_filename.display(),
-                    binary.name
-                );
-            }
+    if std_json {
+        json_contracts.insert(
+            binary.name,
+            JsonContract {
+                abi: abi::ethereum::gen_abi(contract_no, ns),
```

Comment:
Why are we still generating Ethereum ABI?

Reply:
This is for the solidity standard-json format, which is not really used anywhere. It was used by burrow and ewasm, both of which we don't support any more.

We could remove standard-json but I think it might be useful in the future, for example compiling over http. It needs to be redefined to be something more useful, as you can see below it still refers to ewasm.